polynote-kernel/src/main/scala/polynote/kernel/util/KernelContext.scala

```diff
@@ -3,15 +3,74 @@ package polynote.kernel.util
 import java.io.File
 import java.net.URL
 
+import polynote.kernel.{StringRepr, ValueRepr}
+import polynote.messages.truncateTinyString
+
 import scala.reflect.internal.util.{AbstractFileClassLoader, BatchSourceFile}
 import scala.reflect.internal.util.ScalaClassLoader.URLClassLoader
 import scala.reflect.io.{AbstractFile, VirtualDirectory}
+import scala.reflect.runtime
+import scala.reflect.runtime.universe
 import scala.tools.nsc.Settings
 import scala.tools.nsc.interactive.Global
+import scala.tools.reflect.ToolBox
+
+final case class KernelContext(global: Global, classPath: List[File], classLoader: AbstractFileClassLoader) {
+  import global.{Type, Symbol}
+
+  val runtimeMirror: universe.Mirror = scala.reflect.runtime.universe.runtimeMirror(classLoader)
+
+  val importToRuntime: Importer[runtime.universe.type, global.type] = runtime.universe.internal.createImporter(global)
+
+  val importFromRuntime: Importer[global.type, runtime.universe.type] = global.internal.createImporter(scala.reflect.runtime.universe)
+
+  val runtimeTools: ToolBox[universe.type] = runtimeMirror.mkToolBox()
+
+  def inferType(value: Any): Type = {
```

Comment:
Hmm, should this be `global.Type`? would it make a difference?

Reply:
`global.Type` is imported, so it's the same. But I'll change it anyway.

Reply:
Fixed